### ISSUE TITLE
Deallocate is not always correctly printed in Return statement

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#950]
**Describe the bug**
When the returned variable is assigned in an if block, deallocations are not printed in `_print_Return`

**To Reproduce**
Provide code to reproduce the behavior:
```python
def f(b: bool):
    import numpy as np
    x = np.empty(4)
    if b:
        n = 2
    else:
        n = 1

    return n
```

Provide the generated code, or the error message:
```c
#include "tmp.h"
#include <stdlib.h>
#include <stdint.h>
#include <stdbool.h>
#include "ndarrays.h"

/*........................................*/
int64_t f(bool b)
{
    t_ndarray x;
    int64_t n;
    x = array_create(1, (int64_t[]){4}, nd_double);
    if (b)
    {
        n = 2;
    }
    else
    {
        n = 1;
    }
    return n;
}
/*........................................*/
```

**Expected behavior**
Arrays should be deallocated

**Language**
C

**Comment by EmilyBourne** [comment]
When fixing `Return`, please also remove the unnecessary `\n` in [`_print_Return`](https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/ccode.py#L1355)

**Comment by Pinkyboi** [comment]
Hello, when testing the code provided i couldn't reproduce the issue, there's effectively a call to `free_array` before the return.

**Comment by EmilyBourne** [comment]
> Hello, when testing the code provided i couldn't reproduce the issue, there's effectively a call to `free_array` before the return.

I agree. This bug appears to be fixed as of v1.4.0
